PurchasesCoreSwift/Public/Purchases.swift

```diff
@@ -0,0 +1,1275 @@
+                  offeringsManager: offeringsManager,
+                  purchasesOrchestrator: purchasesOrchestrator)
+    }
+
+    @objc
+    public init(appUserID: String?,
+                requestFetcher: StoreKitRequestFetcher,
+                receiptFetcher: ReceiptFetcher,
+                attributionFetcher: AttributionFetcher,
+                attributionPoster: AttributionPoster,
+                backend: Backend,
+                storeKitWrapper: StoreKitWrapper,
+                notificationCenter: NotificationCenter,
+                systemInfo: SystemInfo,
+                offeringsFactory: OfferingsFactory,
+                deviceCache: DeviceCache,
+                identityManager: IdentityManager,
+                subscriberAttributesManager: SubscriberAttributesManager,
+                operationDispatcher: OperationDispatcher,
+                introEligibilityCalculator: IntroEligibilityCalculator,
+                receiptParser: ReceiptParser,
+                purchaserInfoManager: PurchaserInfoManager,
+                productsManager: ProductsManager,
+                offeringsManager: OfferingsManager,
+                purchasesOrchestrator: PurchasesOrchestrator) {
+
+        Logger.debug(Strings.configure.debug_enabled)
+        Logger.debug(String(format: Strings.configure.sdk_version, Self.frameworkVersion))
+        Logger.user(String(format: Strings.configure.initial_app_user_id, appUserID ?? "nil appUserID"))
+
+        self.requestFetcher = requestFetcher
+        self.receiptFetcher = receiptFetcher
+        self.attributionFetcher = attributionFetcher
+        self.attributionPoster = attributionPoster
+        self.backend = backend
+        self.storeKitWrapper = storeKitWrapper
+        self.offeringsFactory = offeringsFactory
+        self.deviceCache = deviceCache
+        self.identityManager = identityManager
+        self.notificationCenter = notificationCenter
+        self.systemInfo = systemInfo
+        self.subscriberAttributesManager = subscriberAttributesManager
+        self.operationDispatcher = operationDispatcher
+        self.introEligibilityCalculator = introEligibilityCalculator
+        self.receiptParser = receiptParser
+        self.purchaserInfoManager = purchaserInfoManager
+        self.productsManager = productsManager
+        self.offeringsManager = offeringsManager
+        self.purchasesOrchestrator = purchasesOrchestrator
+
+        super.init()
+
+        identityManager.configure(appUserID: appUserID)
+        self.purchasesOrchestrator.maybeDelegate = self
+
+        systemInfo.isApplicationBackgrounded { isBackgrounded in
+            if !isBackgrounded {
+                self.operationDispatcher.dispatchOnWorkerThread {
+                    self.updateAllCaches(completion: nil)
+                }
+            } else {
+                self.purchaserInfoManager.sendCachedPurchaserInfoIfAvailable(appUserID: self.appUserID)
+            }
+        }
```

Comment:
nitpick: I think this would look (very slightly) nicer with the condition inverted, as in: 
```swift
systemInfo.isApplicationBackgrounded { isBackgrounded in
    if  isBackgrounded {
        self.purchaserInfoManager.sendCachedPurchaserInfoIfAvailable(appUserID: self.appUserID)
    } else {
        self.operationDispatcher.dispatchOnWorkerThread {
            self.updateAllCaches(completion: nil)
        }
    }
}
```

or even

```swift
systemInfo.isApplicationBackgrounded { isBackgrounded in
    guard isBackgrounded else {
        self.purchaserInfoManager.sendCachedPurchaserInfoIfAvailable(appUserID: self.appUserID)
        return
    }
    self.operationDispatcher.dispatchOnWorkerThread {
        self.updateAllCaches(completion: nil)
    }
}
```

Reply:
Agreed. Since there are no other things getting executed in this block, a `guard` doesn't really mean much, so I'm going with your first suggestion 😄